src/smt/proof_manager.cpp

```diff
@@ -235,8 +234,11 @@ void PfManager::printProof(std::ostream& out,
   Trace("smt-proof") << "PfManager::printProof: start" << std::endl;
   // We don't want to invalidate the proof nodes in fp, since these may be
   // reused in further check-sat calls, or they may be used again if the
-  // user asks for the proof again (in non-incremental mode).
-  if (mode != options::ProofFormatMode::NONE)
+  // user asks for the proof again (in non-incremental mode). We don't need to
+  // clone if the printing below does not modify the proof, which is the case
+  // for proof formats ALF and NONE.
```

Comment:
The best way to do this would be to make the ALF printer take a const reference to the proof only. I can investigate in a later PR if this is possible.